src/bt/Node.cpp

```diff
@@ -114,4 +114,17 @@ std::string Node::status_print(Node::Status s) {
     }
 }
 
+void Node::setRole(std::string roleName) {
+    auto children = this->getChildren();
+    if (properties) {
```

Comment:
?... Isn't properties a default constructed shared map in every case? why bother checking whether the pointer is null when it (should) never be?